pyfar/classes/filter.py

```diff
@@ -284,14 +284,25 @@ def impulse_response(self, n_samples):
 
         Parameters
         ----------
-        n_samples : int
-            Length in samples for which the impulse response is computed.
+        n_samples : int, None
+            Length in samples for which the impulse response is computed. If
+            this is ``None``the length is estimated using
+            `minimum_impulse_response_length((`.
 
         Returns
         -------
         impulse_response : Signal
             The impulse response of the filter.
```

Comment:
```suggestion
            The impulse response of the filter of with a ``cshape = (C, )``
            where `C` denotes the number of channels of the filter.
```

Reply:
I went for Annes suggestion. n_channels would be more verbose but would still require an explicit definition and I though '``cshape = (n_channeles, )`` where `n_channels` denotes the number of channels of the filter.' would be a little lengthy.

Reply:
I changed this for all `impulse_response` functions.